### PR TITLE
Adds MtG except in space

### PR DESCRIPTION
**Pull request by Sypsoti**
Adds Saints and Sins, a MtG Commander-like game simplified way down. Instructions come at the top of the file as well as on the in-game deck box. Comes with pretty sprites & fun card names!

Marking as DNM because I would like to add a few more booster packs first and then throw it all in cargo, but I need the core code reviewed. Lohikar had suggested decls and this is my first time rlly working with them. Thanks for the help, Lohikar <3

### CODE REVIEW

**Comment by Varlaisvea** [comment]
+10000 speedmer-- wait shit dnm, right
+10000 speedreview

File: code/modules/games/saintsandsins_cards.dm [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+ * Can have special cards not found in builder pack.
+ */
+
+/obj/item/pack/saintsandsins/booster
+	name = "\improper Saints and Sins booster pack"
+	desc = "A booster pack for hit collectable card game, Saints and Sins. This one is labeled 'Basic Witch Kit'."
+	pack_type = CARDPACK_BASICWITCH
+	how_many_cards = 11
+
+/obj/item/pack/saintsandsins/proc/build_card_weights_list(pack_type, root_type)
```
Comment (Lohikar) [inline code/modules/games/saintsandsins_cards.dm:91]: You should probably move this under `/obj/item/pack/saintsandsins/Initialize`, as it's not part of the `/booster` type.

File: code/modules/games/saintsandsins_cards.dm [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+	desc = "A builder pack for hit collectable card game, Saints and Sins. This one will \
+    help any aspiring deck builder get started."
+	icon_state = "card_pack_saintsandsins"
+	decktype = /datum/playingcard/saintsandsins
+	var/pack_type = CARDPACK_BUILDER
+	var/how_many_cards = 21 /// Boosters give less cards than the builder packs.
+
+
+/obj/item/pack/saintsandsins/Initialize()
+	..()
```
Comment (Lohikar) [inline code/modules/games/saintsandsins_cards.dm:59]:
```suggestion
	. = ..()
```

**Comment by Sypsoti** [comment]
Removing DNM, satisfied with this until I've gotten feedback from people actually playing it.

File: code/modules/games/saintsandsins_cards.dm [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
+		CARDPACK_BUILDER = 10,
+		CARDPACK_BASICWITCH = 10,
+		CARDPACK_ALCHEMIST = 10,
+		CARDPACK_FATEBRINGER = 10
+	)
+
+/decl/sas_card/heretic/dutiful_mephistopheles
+	name = "Dutiful Mephistopheles"
+	health = 19
+	ability = "PASSIVE: Reduces Decay damage received by 2 while increasing Holy by 3."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:153]:
```suggestion
	ability = "PASSIVE: Reduces Decay damage taken by 2 while increasing Holy damage taken by 3."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 4]
```diff
@@ -0,0 +1,673 @@
...
+	pack_probability = list(
+		CARDPACK_BASICWITCH = 6
+	)
+
+/decl/sas_card/heretic/woebringer_colette
+	name = "Woebringer Colette"
+	health = 21
+	ability = "ACTIVE: Deal 2 damage of any damage type while taking 2 damage of the same type."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:161]:
```suggestion
	ability = "ACTIVE: Deal 2 damage of any damage type. Take 2 damage of the same type."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 5]
```diff
@@ -0,0 +1,673 @@
...
+	pack_probability = list(
+		CARDPACK_BASICWITCH = 8
+	)
+
+/decl/sas_card/heretic/orpheus_the_blind
+	name = "Orpheus the Blind"
+	health = 18
+	ability = "PASSIVE: When below half health, take half damage per spell, rounded up."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:169]: Take half damage _from_ spells?

File: code/modules/games/saintsandsins_cards.dm [hunk 6]
```diff
@@ -0,0 +1,673 @@
...
+	health = 20
+	ability = "PASSIVE: Immune to damage-over-time spells. When above half health, take 1 Decay damage per round."
+	pack_probability = list(
+		CARDPACK_ALCHEMIST = 7
+	)
+
+/decl/sas_card/heretic/exalted_atomus
+	name = "Exalted Atomus"
+	health = 17
+	ability = "ACTIVE: Declare a damage type at the start of caster's turn. Nullify that damage type for one round."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:185]:
```suggestion
	ability = "ACTIVE: Declare a damage type at the start of your turn. Nullify that damage type until the start of your next turn."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 7]
```diff
@@ -0,0 +1,673 @@
...
+	health = 21
+	ability = "ACTIVE: Flip a coin. Heads, draw a card. Tails, take 1 Holy damage."
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 6
+	)
+
+/decl/sas_card/heretic/slovenly_pritchard
+	name = "Slovenly Pritchard"
+	health = 18
+	ability = "PASSIVE: Nullify Holy damage received. This can only occur once per round."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:202]:
```suggestion
	ability = "PASSIVE: Nullify the first instance of Holy damage received each round."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 8]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_BUILDER = 8,
+		CARDPACK_BASICWITCH = 10,
+		CARDPACK_ALCHEMIST = 10,
+		CARDPACK_FATEBRINGER = 10
+	)
+
+/decl/sas_card/spell/sick_air
+	name = "Sick Air"
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:285]: ~Bro?~

File: code/modules/games/saintsandsins_cards.dm [hunk 9]
```diff
@@ -0,0 +1,673 @@
...
+		CARDPACK_BASICWITCH = 6,
+		CARDPACK_ALCHEMIST = 6,
+		CARDPACK_FATEBRINGER = 6
+	)
+
+///Basic Witch Kit
+
+/decl/sas_card/spell/protection_ward
+	name = "Protection Ward"
+	ability = "Nullifies the next damage from a spell used on its caster."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:371]:
```suggestion
	ability = "Nullifies damage from the next spell targeting the caster."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 10]
```diff
@@ -0,0 +1,673 @@
...
+	name = "Scorn of the Alchemist"
+	ability = "Caster converts the damage type of another spell to one of their choosing."
+	spell_type = REACTION_TYPE
+	pack_probability = list(
+		CARDPACK_ALCHEMIST = 7
+	)
+
+/decl/sas_card/spell/a_glass_darkly
+	name = "A Glass Darkly"
+	ability = "Doubles the damage of the caster's next KEY spell."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:476]:
```suggestion
	ability = "Doubles the damage of the caster's next Key spell."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 11]
```diff
@@ -0,0 +1,673 @@
...
+	name = "Scrawling Spellbook"
+	ability = "All Heretics draw one card."
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_ALCHEMIST = 7
+	)
+
+/decl/sas_card/spell/grotesque_experiment
+	name = "Grotesque Experiment"
+	ability = "Nullifies another Heretic's spell and they draw a card."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:500]:
```suggestion
	ability = "Nullify a spell. Its caster draws a card."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 12]
```diff
@@ -0,0 +1,673 @@
...
+	name = "Skin Rot"
+	ability = "Deals 3 Decay damage to all Heretics."
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_ALCHEMIST = 7
+	)
+
+/decl/sas_card/spell/rosemary_and_thyme
+	name = "Rosemary and Thyme"
+	ability = "Purges any damage-over-time spells on caster."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:516]:
```suggestion
	ability = "Nullifies all damage-over-time spells targeting the caster."
```
Might need to clarify this for spells that target all heretics.

File: code/modules/games/saintsandsins_cards.dm [hunk 13]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_ALCHEMIST = 6
+	)
+
+/decl/sas_card/spell/culling_of_the_firstborn
+	name = "Culling of the Firstborn"
+	ability = "Caster shares spell damage with casting Heretic."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:524]:
```suggestion
	ability = "Caster of the target spell splits the damage with its original target, rounding up."
```
This is slightly more generalized but I think also a little bit clearer.

File: code/modules/games/saintsandsins_cards.dm [hunk 14]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = REACTION_TYPE
+	pack_probability = list(
+		CARDPACK_ALCHEMIST = 7
+	)
+
+/decl/sas_card/spell/blood_boil
+	name = "Blood Boil"
+	ability = "Deals 1 Decay and 2 Fire damage to Heretic."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:532]:
```suggestion
	ability = "Deals 1 Decay and 2 Fire damage to target."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 15]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_ALCHEMIST = 7
+	)
+
+/decl/sas_card/spell/transmogrification
+	name = "Transmogrification"
+	ability = "Deals 2 damage of caster's chosen damage type."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:540]:
```suggestion
	ability = "Caster chooses a damage type. Deals 2 damage of that type to target."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 16]
```diff
@@ -0,0 +1,673 @@
...
+	name = "Divine Fire"
+	ability = "Deal 1 Holy damage and 1 Fire damage to target."
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 8
+	)
+
+/decl/sas_card/spell/cast_down
+	name = "Cast Down"
+	ability = "Nullifies spell and returns it to that caster's hand."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:565]:
```suggestion
	ability = "Nullifies target spell and returns it to its caster's hand."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 17]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = REACTION_TYPE
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 6
+	)
+
+/decl/sas_card/spell/blood_tithe
+	name = "Blood Tithe"
+	ability = "Roll 1d4. Deal that much Water damage to ALL Heretics, including caster."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:573]:
```suggestion
	ability = "Roll 1d4. Deal that much Water damage to ALL Heretics."
```
including caster is implied in other cards.

File: code/modules/games/saintsandsins_cards.dm [hunk 18]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 6
+	)
+
+/decl/sas_card/spell/dire_portent
+	name = "Dire Portent"
+	ability = "Deal 4 Earth damage to target after 3 rounds."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:581]: What happens if the caster dies before this effect occurs?

File: code/modules/games/saintsandsins_cards.dm [hunk 19]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 4
+	)
+
+/decl/sas_card/spell/hatred
+	name = "Hatred"
+	ability = "Share damage received with another Heretic."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:589]:
```suggestion
	ability = "Split damage received with another Heretic, rounding up."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 20]
```diff
@@ -0,0 +1,673 @@
...
+	name = "Unnatural Bonds"
+	ability = "Heal caster and one other Heretic by 3."
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 6
+	)
+
+/decl/sas_card/spell/suffering_poppets
+	name = "Suffering Poppets"
+	ability = "Reflects spell to all other Heretics."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:621]: Dumb q: what happens if you reflect a spell targeting all Heretics? Do the others receive the effects twofold?

File: code/modules/games/saintsandsins_cards.dm [hunk 21]
```diff
@@ -0,0 +1,673 @@
...
+	spell_type = REACTION_TYPE
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 5
+	)
+
+/decl/sas_card/spell/ancient_pact
+	name = "Ancient Pact"
+	ability = "All damage of last received damage type will be healed instead for one round."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:629]:
```suggestion
	ability = "Until the start of your next turn, all damage of last received damage type will be healed instead."
```

File: code/modules/games/saintsandsins_cards.dm [hunk 22]
```diff
@@ -0,0 +1,673 @@
...
+	name = "Rising Tide"
+	ability = "Deal 2 Earth damage to all Heretics. Next round, heal all Heretics by 3."
+	spell_type = KEY_TYPE
+	pack_probability = list(
+		CARDPACK_FATEBRINGER = 6
+	)
+
+/decl/sas_card/spell/trial_of_the_witch
+	name = "Trial of the Witch"
+	ability = "If no damage was done by caster last round, deal 4 Earth damage to target. If there was, deal 1 Wind damage to target."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:645]:
```suggestion
	ability = "If the caster dealt no damage last round, deal 4 Earth damage to target. Else, deal 1 Wind damage to target."
```

**Review by Atermonera — COMMENTED** [review]
Code looks good. Made a pass of card wording/syntax because hey this stuff is actually important for card games!

File: code/modules/games/saintsandsins_cards.dm (hunk 12 above)
Comment (Sypsoti) [reply code/modules/games/saintsandsins_cards.dm:516]: This is specifically for active DoTs, will adjust to be clearer though.

File: code/modules/games/saintsandsins_cards.dm (hunk 18 above)
Comment (Sypsoti) [reply code/modules/games/saintsandsins_cards.dm:581]: Still applicable, just like the DoTs.

File: code/modules/games/saintsandsins_cards.dm (hunk 20 above)
Comment (Sypsoti) [reply code/modules/games/saintsandsins_cards.dm:621]: Yes :)

File: code/modules/games/saintsandsins_cards.dm [hunk 23]
```diff
@@ -182,7 +182,7 @@
 /decl/sas_card/heretic/exalted_atomus
 	name = "Exalted Atomus"
 	health = 17
-	ability = "ACTIVE: Declare a damage type at the start of caster's turn. Nullify that damage type for one round."
+	ability = "ACTIVE: Declare a damage type at the start of your turn. Nullify that damage type until the start of your next turn."
```
Comment (Atermonera) [inline code/modules/games/saintsandsins_cards.dm:185]: This feels ambiguous about the scope of the spells being targeted but I can't express what I think it means succinctly. Does it nullify only the damage targeting the caster, or the damage targeting _any_ heretic?

**Review by Atermonera — APPROVED** [review]
General :+1:, there's one comment that I'm still dubious about but not sure it's a blocker for merging

**Pull request merged**